js/spreadsheet.js

```diff
@@ -36,28 +33,15 @@ var spreadsheet = function(options) {
    * @param {Array} columns
    * @param {Boolean} isFirstRender - defines if it was first render to prepare the data for correct rendering without data changes after changes are made
    */
+
   function prepareData(rows, columns, isFirstRender) {
     var preparedData = rows.map(function(row) {
-      var dataRow = columns.map(function(header, index) {
+      var dataRow = columns.map(function(header) {
         var value = row.data[header];
 
-        if (Array.isArray(value)) {
-          if (arrayColumns.indexOf(header) === -1) {
-            arrayColumns.push(header);
-          }
-
-          // Add double quotes to the string if it contains a comma
-          value = value.map(function(val) {
-            // Stringify value only for the first render for nested arrays
-            if (isFirstRender && value && typeof val !== 'string') {
-              return JSON.stringify(val);
-            }
-
-            return typeof val === 'string' && val.indexOf(',') !== -1 ? '"' + val + '"' : val;
-          }).join(', ');
-        // Stringify value only for the first render for nested objects
-        } else if (isFirstRender && value && typeof value === 'object') {
-          if (objColumns.indexOf(header) === -1) {
+        // Stringify values for rendering objects for the first rendering and when undoing changes to the table (Ctrl + Z in the table cell)
+        if (typeof value === 'object' && value !== null) {
+          if (isFirstRender && objColumns.indexOf(header) === -1) {
             objColumns.push(header);
           }
 
```

Comment:
The next line `value = JSON.stringify(value);` used to only execute if `isFirstRender` is true. Now it's executed even if it isn't `true`. Is that intentional and if so, why is it necessary?

Reply:
I found another bug in the table. I checked, it's reproduced on the staging. 

1. Edit the object.
2. Select another cell to save changes to the object.
3. Select the cell from the object again and press backspace
4. Press Ctrl + Z to return the object.
All objects in the table will become [Object object]

Reply:
@OleksandrLavrynenko 👍🏻  Can you add a comment in the Jira ticket and mention this bug please, including the steps to replicate.

cc @inna-bieshulia